src/components/AudioPlayer/AudioPlayer.tsx

```diff
@@ -52,12 +52,17 @@ const AudioPlayer = () => {
   const audioPlayerRef = useRef<HTMLAudioElement>();
   const audioService = useContext(AudioPlayerMachineContext);
   const isVisible = useSelector(audioService, (state) => state.matches('VISIBLE'));
+  const volume = useSelector(audioService, (state) => state.context.volume);
 
   useEffect(() => {
     window.audioPlayerEl = audioPlayerRef.current;
     audioService.send({ type: 'SET_AUDIO_REF', audioPlayerRef: audioPlayerRef.current });
   }, [audioService]);
 
+  useEffect(() => {
+    audioPlayerRef.current.volume = volume;
+  }, [volume, audioPlayerRef]);
+
```

Comment:
This is un-necessary since the `audioPlayerMachine` is in control of the audio player and should set the volume of the audio player from inside an action rather than a React component. You can take a look at usage examples of `context.audioPlayer` inside `src/xstate/actors/audioPlayer/audioPlayerMachine.ts`. E.g. `pauseAudio` action.